### PR TITLE
Added sprinting and slow walk

### PR DESCRIPTION
**Pull request by Lurk390**
Resolves issue #97, resolves issue #98

- Added sprinting by double tapping the forward (`W`) button
- Added slow walk (crouching) by holding the `shift` button while moving
- Added tests to test the logic and implementation of both features

### CODE REVIEW

File: tempus_fugit_minecraft/player.py [hunk 1]
```diff
@@ -291,3 +291,25 @@ def keep_player_within_coordinates(dimension, boundary_size=WORLD_SIZE):
...
+        """!
+        @brief Resets the player's walking speed to the default value
+        @see [Issue#97](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/97)
+        @see [Issue#98](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/98)
+        """
+        self.walking_speed = 5
+
+    def sprint(self) -> None:
+        """!
+        @brief Increases the player's walking speed
```
Comment (w1158mxs) [inline tempus_fugit_minecraft/player.py:312]: I think increasing walking should use (add and sub) functions in the calculation, while you using multiplication you need to name it double speed because I see the function doubling speed not increasing if you want to be more specific.

**Review by w1158mxs — CHANGES_REQUESTED** [review]
I'm wondering what if players walk then stop then walk again! 
if they do it fast it will be like he clicking double-click!

But it's an idea I'll try it.
just do tests nonstop with a decrease and its good to go

**Review by w1158mxs — APPROVED** [review]
your pull request will make conflict with 106

File: tests/test_player.py [hunk 2]
```diff
@@ -407,3 +407,21 @@ def test_player_in_y_coordinate(self, player: Player):
         player.position = (79 , (WORLD_SIZE+1000) , 0)
         player.check_player_within_world_boundaries()
         assert player.position == (79 , (WORLD_SIZE+1000) , 0)
+
+    #issue97
```
[the remark this answers is not in the document]
Comment (Lurk390) [reply tests/test_player.py:411]: I just kept it the same format as the rest of the test. If we want to update our test comments, we should make an issue and address them all.

**Comment by Lurk390** [comment]
> I'm wondering what if players walk then stop then walk again! if they do it fast it will be like he clicking double-click!
> 
> But it's an idea I'll try it. just do tests nonstop with a decrease and its good to go

That's why there is a 0.5 sec window, and it can be adjusted if you think it needs to be changed

File: tempus_fugit_minecraft/player.py [hunk 3]
```diff
@@ -291,3 +291,25 @@ def keep_player_within_coordinates(dimension, boundary_size=WORLD_SIZE):
             return -boundary_size
         else:
             return dimension
+
+    def slow_walk(self) -> None:
```
Comment (xieWright) [inline tempus_fugit_minecraft/player.py:295]: Change name of function to slow_player_walk_speed

File: tempus_fugit_minecraft/player.py [hunk 4]
```diff
@@ -291,3 +291,25 @@ def keep_player_within_coordinates(dimension, boundary_size=WORLD_SIZE):
...
+        """!
+        @brief Slows the player's walking speed
+        @see [Issue#97](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/97)
+        """
+        self.walking_speed = self.walking_speed / 3
```
Comment (xieWright) [inline tempus_fugit_minecraft/player.py:300]: Probably something that needs to be done in a different pull request, but we probably have to change walking_speed to be more specific and include units.

File: tempus_fugit_minecraft/player.py [hunk 5]
```diff
@@ -291,3 +291,25 @@ def keep_player_within_coordinates(dimension, boundary_size=WORLD_SIZE):
...
+
+    def reset_walk_speed(self) -> None:
```
Comment (xieWright) [inline tempus_fugit_minecraft/player.py:302]: Probably need to change function name to something like reset_player_walking_speed_to_default

File: tempus_fugit_minecraft/player.py [hunk 6]
```diff
@@ -291,3 +291,25 @@ def keep_player_within_coordinates(dimension, boundary_size=WORLD_SIZE):
...
+        """!
+        @brief Resets the player's walking speed to the default value
+        @see [Issue#97](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/97)
+        @see [Issue#98](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/98)
+        """
+        self.walking_speed = 5
+
+    def sprint(self) -> None:
```
Comment (xieWright) [inline tempus_fugit_minecraft/player.py:310]: Also another name change. Probably make_player_sprint or something.

**Review by xieWright — CHANGES_REQUESTED** [review]
Mostly just requesting name changes. Professor was getting up in our grill about that.

File: tempus_fugit_minecraft/player.py (hunk 3 above)
Comment (Lurk390) [reply tempus_fugit_minecraft/player.py:295]: Sure, I will change it to `slow_walking_speed` since it's already in the `Player` class.

**Review by criticalglitch — CHANGES_REQUESTED** [review]
Please rebase your changes, you have conflicts. Sorry about that.

**Review by criticalglitch — APPROVED** [review]
Here you are.

**Review by xieWright — APPROVED** [review]
Name changes are there. Passes the auto tests and hopefully the name changes fulfill the literate programming requirement.
